jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/dao/RDSDBManager.java

```diff
@@ -207,4 +213,43 @@ public static void insertRecord(String tableName, Map<String, String> data) {
             LOGGER.error("Error Executing Query", e);
         }
     }
+
+    public static boolean isFirstTimePluginExecuted(String dataSource, List<PolicyTable> policyList) {
+        String recordsCountQuery = "SELECT COUNT(*) AS count FROM cf_PolicyTable WHERE assetGroup = ?";
+        try (Connection conn = getConnection();
+             PreparedStatement preparedStatement = conn.prepareStatement(recordsCountQuery)) {
+            preparedStatement.setString(1, dataSource);
+            return executeCountQuery(preparedStatement) == 0;
+        } catch (SQLException sqlException) {
+            LOGGER.error("Error executing query", sqlException);
+        }
+        return false;
+    }
+
+    private static String determinePolicyStatus(PolicyTable policy, boolean hasPluginBeenExecutedBefore) {
+        if (hasStatus(policy.getStatus())) { //Case 1 : When Policy has Status (Contrast) use that
+            return policy.getStatus();
+        }
+        if (hasPluginBeenExecutedBefore) { //Case 2 : When Policy Status is NULL (Checkmarx/Wiz) and Plugin has been executed before
+            return isPolicyNew(policy)? "DISABLED":"";//Case 2.a : If Policy is new, set status to DISABLED .IGNORE UPDATE operation
+        }
+        return "ENABLED";   //Case 3 : When Policy Status is NULL (Checkmarx/Wiz) and Plugin is being executed for the first time
+    }
+
+    public static boolean isPolicyNew(PolicyTable policy) {
+        return !policyIdsSet.contains(policy.getPolicyId());
+    }
+
+    private static int executeCountQuery(PreparedStatement preparedStatement) throws SQLException {
+        try (ResultSet rs = preparedStatement.executeQuery()) {
+            if (rs.next()) {
+                return rs.getInt("count");
+            }
+        }
+        return 0;
+    }
+
+    private static boolean hasStatus(String policyStatus) {
+        return !StringUtils.isNullOrEmpty(policyStatus);
```

Comment:
this is just a wrapper method of StringUtils.isNullOrEmpty. I'd suggest adding this condition inline instead of creating this method

Reply:
Removed this method and used inline